roles/pulp-routes/tasks/main.yml

```diff
@@ -1,8 +1,30 @@
 ---
+- name: Load Route TLS certificate
+  include_tasks: load_route_tls_secret.yml
+  when:
+    - ingress_type | lower == 'route'
+    - route_tls_secret != ''
+
+- set_fact:
+    _pulp_plugins:
+      - { name: "{{ ansible_operator_meta.name }}-content", path: "/pulp/content/", targetPort: "content-24816", serviceName: "{{ ansible_operator_meta.name }}-content-svc" }
+      - { name: "{{ ansible_operator_meta.name }}-container", path: "/pulp/container/", targetPort: "content-24816", serviceName: "{{ ansible_operator_meta.name }}-content-svc" }
+      - { name: "{{ ansible_operator_meta.name }}-cookbook", path: "/pulp_cookbook/content/", targetPort: "content-24816", serviceName: "{{ ansible_operator_meta.name }}-content-svc" }
+      - { name: "{{ ansible_operator_meta.name }}", path: "/", targetPort: "api-24817", serviceName: "{{ ansible_operator_meta.name }}-api-svc" }
+      - { name: "{{ ansible_operator_meta.name }}-api-v3", path: "/pulp/api/v3/", targetPort: "api-24817", serviceName: "{{ ansible_operator_meta.name }}-api-svc" }
+      - { name: "{{ ansible_operator_meta.name }}-auth", path: "/auth/login/", targetPort: "api-24817", serviceName: "{{ ansible_operator_meta.name }}-api-svc" }
+      - { name: "{{ ansible_operator_meta.name }}-v2", path: "/v2/", targetPort: "api-24817", serviceName: "{{ ansible_operator_meta.name }}-api-svc" }
+      - { name: "{{ ansible_operator_meta.name }}-extensions-v2", path: "/extensions/v2/", targetPort: "api-24817", serviceName: "{{ ansible_operator_meta.name }}-api-svc" }
+      - { name: "{{ ansible_operator_meta.name }}-token", path: "/token/", targetPort: "api-24817", serviceName: "{{ ansible_operator_meta.name }}-api-svc" }
+      - { name: "{{ ansible_operator_meta.name }}-python", path: "/pypi/", targetPort: "api-24817", serviceName: "{{ ansible_operator_meta.name }}-api-svc" }
+      - { name: "{{ ansible_operator_meta.name }}-ansible", path: "/pulp_ansible/galaxy/", targetPort: "api-24817", serviceName: "{{ ansible_operator_meta.name }}-api-svc" }
```

Comment:
I'm wondering if we can build a parser that gets the plugin nginx snippets and converts into this "notation", so we wouldn't need to manually add new entries when they are updated

Reply:
I tried:
```bash
pip install python-nginx
```
```python
import json
import nginx

nginx_configs = ["pulp_container/app/webserver_snippets/nginx.conf"]
name = "{{ ansible_operator_meta.name }}"
router = []

for plugin_conf in nginx_configs:
    app = "container"
    conf = nginx.loadf(plugin_conf)
    for location in conf.filter("Location"):
        path = location.value
        if not path.startswith("/"):
            path = f"/{path}"
        for key in location.keys:
            target_port = ""
            svc_name = ""
            if key.name.strip("'") == "proxy_pass":
                if "pulp-api" in key.value:
                    target_port = "api-24817"
                    svc_name = f"{name}-api-svc"
                    break
                if "pulp-content" in key.value:
                    target_port = "content-24816"
                    svc_name = f"{name}-content-svc"
                    break
        if not svc_name:
            raise RuntimeError(f"Location {path} doesn't have proxy_pass")
        router.append(
            {
                "name":f"{name}-{app}{path.rstrip('/').replace('/', '-')}", 
                "path": path,
                "targetPort": target_port,
                "serviceName": svc_name,
            }
        )

print(json.dumps(router, indent=4))
```
the result:
```json
[
    {
        "name": "{{ ansible_operator_meta.name }}-container-v2",
        "path": "/v2/",
        "targetPort": "api-24817",
        "serviceName": "{{ ansible_operator_meta.name }}-api-svc"
    },
    {
        "name": "{{ ansible_operator_meta.name }}-container-extensions-v2",
        "path": "/extensions/v2/",
        "targetPort": "api-24817",
        "serviceName": "{{ ansible_operator_meta.name }}-api-svc"
    },
    {
        "name": "{{ ansible_operator_meta.name }}-container-pulp-container",
        "path": "/pulp/container/",
        "targetPort": "content-24816",
        "serviceName": "{{ ansible_operator_meta.name }}-content-svc"
    },
    {
        "name": "{{ ansible_operator_meta.name }}-container-token",
        "path": "/token/",
        "targetPort": "api-24817",
        "serviceName": "{{ ansible_operator_meta.name }}-api-svc"
    }
]
```

Reply:
https://github.com/pulp/pulp-operator/issues/489